pkg/apis/batch/validation/validation.go

```diff
@@ -486,9 +499,63 @@ func validatePodTemplateUpdate(spec, oldSpec batch.JobSpec, fldPath *field.Path,
 }
 
 // ValidateJobStatusUpdate validates an update to a JobStatus and returns an ErrorList with any errors.
-func ValidateJobStatusUpdate(status, oldStatus batch.JobStatus) field.ErrorList {
+func ValidateJobStatusUpdate(job, oldJob *batch.Job) field.ErrorList {
```

Comment:
Out of pure curiosity, have you tried running this with a regular job controller and ensure this is matching the expectations of these validations?

Reply:
This is now Beta (enabled by default), so it is run during all integration and e2e tests, I think they would fail if the validation would fail.

Reply:
Marking as resolved to focus attention of the remaining questions. Feel free to open a new thread if related to that.